lib/api/contracts/model_contract.rb

```diff
@@ -42,12 +42,27 @@ def self.attribute_validations
       end
 
       def self.attribute(*attributes, &block)
+        attributes.each do |attribute|
+          property attribute
+        end
+
         writable_attributes.concat attributes.map(&:to_s)
         if block
           attribute_validations << block
         end
       end
 
+      # we want to add a validation error whenever someone sets a property that we don't know.
+      # However AR will cleverly try to resolve the value for errorneous properties. Thus we need
+      # to hook into this method and return nil for unknown properties to avoid NoMethod errors...
+      def read_attribute_for_validation(attribute)
+        if respond_to? attribute
+          send attribute
+        end
+
+        nil
```

Comment:
Doesn't this always return nil, even if `attribute` exists in the Model? I'm not entirely sure what this method should do, as the original is [just an alias to send](https://github.com/rails/rails/blob/fe664e8ce97b5f7a1e1d95dd4df0fbdba42c7f2b/activemodel/lib/active_model/validations.rb#L390), but I assume nil should be in an else block?